src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -472,6 +575,43 @@ private ICredentialProvider GetCredentialProvider(CommandSettings command, strin
             return provider;
         }
 
+        private void AssertAdminAccess(bool unConfigure = false)
+        {
+            if (new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator))
+            {
+                return;
+            }
+
+            if(unConfigure)
+            {
+                Trace.Error("Needs Administrator privileges to unconfigure an agent running with autologon capability.");          
+                throw new SecurityException(StringUtil.Loc("NeedAdminForAutologonRemoval"));                
+            }
+            else
+            {
+                Trace.Error("Needs Administrator privileges for configure agent as interactive process with autologon capability.");
+                Trace.Error("You will need to unconfigure the agent and then re-configure with Administrative rights");            
+                throw new SecurityException(StringUtil.Loc("NeedAdminForAutologonCapability"));
+            }
+        }        
+        private async Task TestConnectAsync(string url, VssCredentials creds)
```

Comment:
thanks.

---
In reply to: [117341366](https://github.com/Microsoft/vsts-agent/pull/978#discussion_r117341366) [](ancestors = 117341366)